localization/landmark_based_localizer/ar_tag_based_localizer/src/ar_tag_based_localizer.cpp

```diff
@@ -182,18 +183,13 @@ void ArTagBasedLocalizer::image_callback(const Image::ConstSharedPtr & msg)
   Pose self_pose;
   {
     // get self-position on map
-    std::unique_lock<std::mutex> self_pose_array_lock(self_pose_array_mtx_);
-    if (self_pose_msg_ptr_array_.size() <= 1) {
-      RCLCPP_WARN_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 1, "No Pose!");
+    const std::optional<SmartPoseBuffer::InterpolateResult> interpolate_result =
+      ekf_pose_buffer_->interpolate(sensor_stamp);
+    if (!interpolate_result) {
       return;
     }
-    PoseArrayInterpolator interpolator(
-      this, sensor_stamp, self_pose_msg_ptr_array_, ekf_time_tolerance_, ekf_position_tolerance_);
-    if (!interpolator.is_success()) {
-      return;
-    }
-    pop_old_pose(self_pose_msg_ptr_array_, sensor_stamp);
-    self_pose = interpolator.get_current_pose().pose.pose;
+    ekf_pose_buffer_->pop_old(sensor_stamp);
+    self_pose = interpolate_result.value().interpolated_pose.pose.pose;
   }
```

Comment:
I guess that this scope was for mutex. Now that pose_buffer is thread-safe, I think we can eliminate this scope (`{ }`) and change the `self_pose` to const variable.
`const Pose self_pose=interpolate_result.value().interpolated_pose.pose.pose;`

Reply:
Very good point. Completely agree.
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/5713/commits/4b5038778981129c2a2475300b9431db4384dfa0